### ISSUE TITLE
Should the named arguments being on separate lines be considered a lint?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1042]
Here is an example code:

```r
mean(
  x = 
    c(1, 2, 3, NA, 5),
  na.rm =
    TRUE
)
```

The _Tidyverse style guide_ says [nothing](https://style.tidyverse.org/syntax.html#argument-names) about whether the parameter name and the argument provided to it should be on the same lines:

<img width="748" alt="Screenshot 2022-04-02 at 10 28 42" src="https://user-images.githubusercontent.com/11330453/161374541-e33aec9e-2b95-4f21-a11d-0c98b94b5268.png">

And `{styler}` will [not change](https://github.com/r-lib/styler/issues/734) styling for this code:

``` r
styler::style_text(
  "mean(
  x =
    c(1, 2, 3, NA, 5),
  na.rm =
    TRUE
  )"
)
#> mean(
#>   x =
#>     c(1, 2, 3, NA, 5),
#>   na.rm =
#>     TRUE
#> )
```

<sup>Created on 2022-04-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

But I am wondering how intentional such a usage might be in the wild, and if this should be considered a lint?

**Comment by AshesITR** [comment]
Thanks for the suggestion.

There [actually](https://style.tidyverse.org/syntax.html#long-lines) is a corresponding rule in the styleguide: In long calls, every argument should go on a single line.
The only unmentioned case is what to do if the argument doesn't fit a single line (I assume it should then be factored into a separate variable, like `x <- very_long_computation; mean(x = x)`)

Feel free to submit a PR.

**Comment by MichaelChirico** [comment]
FWIW we use this for consistency in cases like

```
call(
  key1 =
    short_expr,
  key2 =
    a_very_long_expr_that_violates_80_chars
)
```

e.g. this is very common for mutate() steps with many columns

**Comment by IndrajeetPatil** [comment]
Thanks for the example, Michael.

I think it makes a strong case that this coding style should _not_ be considered a lint, since there are good reasons for people not have arguments on the same line as the named parameters.
